### PR TITLE
Closing a PeerConnection should not fire iceconnectionstatechange event.

### PR DESCRIPTION
**Pull request by jianjunz**
Add a test case for w3c/webrtc-pc#2335 and w3c/webrtc-pc#2336.

### CODE REVIEW

File: webrtc/RTCPeerConnection-iceConnectionState.https.html
Comment (jan-ivar) [inline webrtc/RTCPeerConnection-iceConnectionState.https.html]: We should use `exchangeIceCandidates` here. There's no spec support for requiring ice candidates be queued to avoid intermittents. That's a Chrome bug that should not dictate wpt tests. But I see this is prevalent, so I've opened https://github.com/web-platform-tests/wpt/issues/19866  for that.

**Review by jan-ivar — APPROVED** [review]

File: webrtc/RTCPeerConnection-iceConnectionState.https.html
Comment (alvestrand) [reply webrtc/RTCPeerConnection-iceConnectionState.https.html]: We should use coupleIceCandidates until we have a test that solidly exercises the case that causes the issue.

Comment (jan-ivar) [reply webrtc/RTCPeerConnection-iceConnectionState.https.html]: Agree. The fiddle in https://github.com/web-platform-tests/wpt/issues/19866#issuecomment-546114464 should make a good test.

Comment (jianjunz) [reply webrtc/RTCPeerConnection-iceConnectionState.https.html]: Thanks for your review. `coupleIceCandidates` has been changed to `exchangeIceCandidates`.

Comment (jianjunz) [reply webrtc/RTCPeerConnection-iceConnectionState.https.html]: Updated this PR to address the comment above. Could you please take a look and help merge it? Thanks.

Comment (jan-ivar) [reply webrtc/RTCPeerConnection-iceConnectionState.https.html]: Lgtm

File: webrtc/RTCPeerConnection-iceConnectionState.https.html
Comment (henbos) [inline webrtc/RTCPeerConnection-iceConnectionState.https.html]: Optional suggestion: Add this line before asserting:
await Promise.resolve();

This might catch an implementation that surfaces the invalid iceConnectionState change in a posted task

**Review by henbos — APPROVED** [review]
LGTM with optional suggestion

File: webrtc/RTCPeerConnection-iceConnectionState.https.html
Comment (jan-ivar) [reply webrtc/RTCPeerConnection-iceConnectionState.https.html]: Good point, or the test may end prematurely. Probably even a t.set_timeout or whatever it is called.

Comment (jianjunz) [reply webrtc/RTCPeerConnection-iceConnectionState.https.html]: Thanks for your suggestion. A new patch was submitted. Please help merge this PR if it looks good to you. I don't have write access to merge it. Thanks.

**Review by henbos — APPROVED** [review]

File: webrtc/RTCPeerConnection-iceConnectionState.https.html
Comment (jan-ivar) [inline webrtc/RTCPeerConnection-iceConnectionState.https.html]: Actually, we want to assert *first*, to test the state is closed synchronously, followed by allowing sufficient time to ensure an event does not come in.

Unfortunately, `await Promise.resolve()` only queues a microtask, which merely defers execution to the end of *this* run of the main event loop. This is not enough time for an event to have fired, since the firing of this event would be queued in a new task. We need:
```js
await new Promise(r => t.step_timeout(r));
```

Comment (jianjunz) [reply webrtc/RTCPeerConnection-iceConnectionState.https.html]: Yes, ICE connection state should be set to "closed" synchronously. I'm going to submit a new PR to fix it and wait 100ms for `oniceconnectionstatechange` event.

**Pull request merged**
